packages/cli/src/WaitTracker.ts

```diff
@@ -26,7 +27,9 @@ export class WaitTracker {
 		private readonly executionRepository: ExecutionRepository,
 		private readonly ownershipService: OwnershipService,
 		private readonly workflowRunner: WorkflowRunner,
+		private readonly orchestrationService: OrchestrationService,
 	) {
+		if (this.orchestrationService.isFollower) return;
```

Comment:
What if this instance becomes the leader? It should subscribe to the `leadershipChange` event and start checking the database.

Similarly, if a leader is for whatever reason demoted (such as dropping the ball), it should also stop polling.

Reply:
Done, please see [7419fe9](https://github.com/n8n-io/n8n/pull/9014/commits/7419fe95ec19a255521bcde450ba34b4a87bc609). 

Note that `PruningService` is unaware of orchestration, but for `WaitTracker` I had to set up orchestration in the constructor because the two services are initialized differently, i.e. `PruningService` is initialized explicitly at `start.ts` but `WaitTracker` inits only as a dependency of `ActiveExecutionService`, which inits only as a dependency of `ExecutionsController`, which inits when the server is set up. My aim was to avoid altering this sequence to prevent side effects, but please let me know if there's a better approach.

Reply:
Were you able to test this? I'm afraid we might have a dependency cycle, but it might simply not happen. If you were able to start multi main with a couple mains and the process worked fine, then it should be all good. I think with the current setup, if there's a dependency cycle issue it should be visible immediately.